sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -218,6 +237,92 @@ void gnrc_rpl_recv_DIS(gnrc_rpl_dis_t *dis, ipv6_addr_t *src, ipv6_addr_t *dst,
     }
 }
 
+static bool _gnrc_rpl_check_options_validity(int msg_type, gnrc_rpl_dodag_t *dodag,
```

Comment:
I thought about this, but this would mean that options that have been parsed already cannot be undone. This is especially harmful if there are options which relate to each other. This early check discards the whole packet before parsing it. You are right that it is kinda inefficient, but how many options would be included in a control message. 2-4?